string/location.ts

```diff
@@ -0,0 +1,21 @@
+import { TextLocation, TextWithLocation } from "./types.ts";
+
+export function location(text: string, index: number): TextLocation {
+  const linesText = text.slice(0, index).split("\n");
+
+  const line = linesText.length;
+  const column = linesText[linesText.length - 1].length + 1;
+
+  const location = { offset: index, line: line, column: column };
+  return location;
```

Comment:
We can return this immediately to remove a line without diminishing legibility. We can also use [object property shorthand syntax](https://eslint.org/docs/latest/rules/object-shorthand) to remove some noise.

```ts
return { offset: index, line, column }
```

Reply:
Agreed